src/manageiq_client/api.py

```diff
@@ -622,6 +622,15 @@ def _process_result(self, result):
             if "id" in result:
                 d["href"] = "{}/{}".format(collection._href, result["id"])
             return Entity(collection, d)
+        elif "task_href" in result or "task_id" in result:
+            collection = getattr(self.api.collections, "tasks")
+            # reuse task_href and task_id, no other data is relevant
+            d = {}
+            if result.get("task_href"):
```

Comment:
Thanks for review @cben 

The `Entity` is returned also when `href`, `id` or `requester_id` keys are present in the response. The message is lost (not really, see below) in all these cases. In this sense my change just extends the existing behavior.

Two remarks about the message:
1. in case of "task", the message returned in the response is not that important. All the up-to-date information is available in the corresponding task entity.

E.g. response
```
'message': "Reverting to snapshot snpshot_Dic0TrKu for Virtual Machine id:1761 name:'test-snpsh-cv1i'"
```
vs info available in the corresponding task entity
```
"name": "Reverting to snapshot snpshot_Dic0TrKu for Virtual Machine id:1761 name:'test-snpsh-cv1i'",
"message": "Task completed successfully",
[...]
```

2. the original response is still available through ``api.response``. The message can be accessed this way -  ``api.response.json()['message']``

Regarding ``if "task_href" in result:`` vs ``result.get("task_href")`` I agree that the former is more consistent with surrounding code. I like the latter better because it tests both membership and value (I guess it's unlikely that the value of "task_href" is false, but this seems more robust to me). Actually I was tempted to change the other code as well :)
In any case I have no problem with changing it to ``if "task_href" in result:``.